src/linux/Packaging.Linux/install-from-source.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+SUDO_CMD=sudo
+
+# Parse script argument that allows users to customize sudo command
+for i in "$@"
+do
+    case "$i" in
+        --sudo-cmd=*)
+        SUDO_CMD="${i#*=}"
+        shift # past argument=value
+        ;;
+    esac
+done
+
+install_shared_packages() {
+    pkg_manager=$1
+    install_verb=$2
+
+    local shared_packages="git curl apt-transport-https"
+    for package in $shared_packages
+    do
+        if [ $pkg_manager = apk ]; then
+            $SUDO_CMD $pkg_manager $install_verb $package
+        else
+            $SUDO_CMD $pkg_manager $install_verb $package -y
+        fi
+    done
+}
+
+download_dotnet_script() {
+    curl -o dotnet-install.sh -L https://dot.net/v1/dotnet-install.sh
+    chmod +x ./dotnet-install.sh
+}
+
+distribution=$(sed -n '/^ID=/p' /etc/os-release | cut -c 4- | tr -d '"')
+case $distribution in
+    debian | ubuntu | linuxmint)
+        # add dotnet package repository/signing key
+        $SUDO_CMD apt update && $SUDO_CMD apt install wget -y
+        eval wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+        $SUDO_CMD dpkg -i packages-microsoft-prod.deb
+        rm packages-microsoft-prod.deb
+
+        # proactively install tzdata to prevent prompts
+        $SUDO_CMD apt update
+        export DEBIAN_FRONTEND=noninteractive
+        $SUDO_CMD apt-get install -y --no-install-recommends tzdata
+
+        install_shared_packages apt install
+
+        # install dotnet packages
+        $SUDO_CMD apt install dotnet-sdk-5.0 dpkg-dev -y
+    ;;
+    fedora | centos | rhel)
+        if [ $distribution = centos ]; then
+            sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-Linux-*
+            sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-Linux-*
```

Comment:
I am not sure that CentOS users would like their `/etc/yum.repos.d/` being edited as part of installing GCM. Can we avoid that?

Reply:
I can try. I've admittedly had a fair amount of trouble with CentOS, but you're right - there's probably a better way.

Reply:
There was nothing in the mirrorlist when I ran this on GH agents. I've updated this to only execute in CI scenarios in my next iteration. Just cleaning things up and will push shortly.

Reply:
Maybe this is a bug in the CI agents (or the Docker container?) that should be handled at a separate level, e.g. in the GitHub workflow?